builds/e2e/compare-compatibility.yaml

```diff
@@ -0,0 +1,188 @@
+# This YAML runs compare script to make sure the current iotedge binaries and containers are in line with IoT Edge Platform Compatibility Script. The conditions being evaluated are Memory Usage, Binary Size Container Size, Shared Libraries, Docker API Version, Kernel Configuration.
+trigger: none
+pr: none
+
+resources:
+  pipelines:
+  - pipeline: images
+    source: 'Azure-IoT-Edge-Core Build Images'
+    branch: 'main'
+    trigger:
+      branches:
+      - main
+      - release/*
+  - pipeline: packages
+    source: 'Azure-IoT-Edge-Core Edgelet Packages'
+    branch: 'main'
+    trigger:
+      branches:
+      - main
+      - release/*
+
+
+jobs:
+
+
+################################################################################
+  - job: ubuntu_2004_msmoby
+################################################################################
+    displayName: Ubuntu 20.04 with iotedge-moby
+
+    pool:
+      name: $(pool.linux.name)
+      demands:
+      - ImageOverride -equals agent-aziotedge-ubuntu-20.04-msmoby
+
+    variables:
+        os: linux
+        arch: amd64
+        artifactName: iotedged-ubuntu20.04-amd64
+        identityServiceArtifactName: packages_ubuntu-20.04_amd64
+        identityServicePackageFilter: aziot-identity-service_*_amd64.deb
+        skip_notary: true
+    
+    timeoutInMinutes: 180
+
+    steps:
+    - template: templates/e2e-setup.yaml
+    - task: AzureKeyVault@1
+      displayName: 'Get Secret'
+      inputs:
+        azureSubscription: $(az.subscription)
+        KeyVaultName: $(az.keyvault)
+        SecretsFilter: >-
+          edgebuild-service-principal-secret,
+    - task: Bash@3
+      displayName: 'Az login'
```

Comment:
Isn't there an Azure CLI task you can use that takes a service connection, rather than having to store a service principal in key vault?

Reply:
The problem with that is that Azure CLI doesn't exist in ARM64, 

Error : https://dev.azure.com/msazure/One/_build/results?buildId=53304245&view=logs&j=8520a43c-dcec-5ce4-f544-4575e5778c8d&t=3246ff34-2452-5cd6-7065-cdab75bf1c05

In the spirit of keeping things uniform, I opted for using AZ Login. The service principal already existed in the keyvault, I did not add a new entry